bridges/otelzap/zap_encoder.go

```diff
@@ -0,0 +1,233 @@
+	stringValue := strconv.FormatComplex(v, 'f', -1, 64)
+	m.cur = append(m.cur, log.String(k, stringValue))
+}
+
+// Converts Float64 to logFloat64.
+func (m *ObjectEncoder) AddFloat64(k string, v float64) {
+	m.cur = append(m.cur, log.Float64(k, v))
+}
+
+// Converts Float32 to logFloat64.
+func (m *ObjectEncoder) AddFloat32(k string, v float32) {
+	// preserves float32 value
+	value, _ := strconv.ParseFloat(strconv.FormatFloat(float64(v), 'f', -1, 32), 64)
+	m.AddFloat64(k, value)
+}
+
+// Converts Int64 to logInt64.
+func (m *ObjectEncoder) AddInt64(k string, v int64) {
+	m.cur = append(m.cur, log.Int64(k, v))
+}
+
+// Converts Int to logInt.
+func (m *ObjectEncoder) AddInt(k string, v int) {
+	m.cur = append(m.cur, log.Int(k, v))
+}
+
+// Converts String to logString.
+func (m *ObjectEncoder) AddString(k string, v string) {
+	m.cur = append(m.cur, log.String(k, v))
+}
+
+func (m *ObjectEncoder) AddUint64(k string, v uint64) {
+	m.cur = append(m.cur,
+		log.KeyValue{
+			Key:   k,
+			Value: assignUintValue(v),
+		})
+}
+
+// It calls this method if interface cannot be mapped to supported zap types
+// converts everything to a JSON string.
+func (m *ObjectEncoder) AddReflected(k string, v interface{}) error {
+	enc := json.NewEncoder(m)
+	if err := enc.Encode(v); err != nil {
+		return err
+	}
+	m.AddString(k, m.reflectval.AsString())
+	return nil
+}
+
+// Implements Write method to which json encoder writes to.
+// Used by AddReflected method.
+func (m *ObjectEncoder) Write(p []byte) (n int, err error) {
+	m.reflectval = log.StringValue(string(p))
+	return
+}
+
+// TODO:
+// OpenNamespace opens an isolated namespace where all subsequent fields will
+// be added.
+func (m *ObjectEncoder) OpenNamespace(k string) {
+	// ns := make([]log.KeyValue, 0)
+	// // m.cur expects both key and value
+	// // "Namespace" as value here should be confirmed
+	// m.cur = append(m.cur, log.String(k, "Namesspace"))
+	// m.cur = ns
+}
+
+func (m *ObjectEncoder) AddComplex64(k string, v complex64) { m.AddComplex128(k, complex128(v)) }
+func (m *ObjectEncoder) AddInt32(k string, v int32)         { m.AddInt64(k, int64(v)) }
+func (m *ObjectEncoder) AddInt16(k string, v int16)         { m.AddInt64(k, int64(v)) }
+func (m *ObjectEncoder) AddInt8(k string, v int8)           { m.AddInt64(k, int64(v)) }
+func (m *ObjectEncoder) AddTime(k string, v time.Time)      { m.AddInt64(k, v.UnixNano()) }
+func (m *ObjectEncoder) AddUint(k string, v uint)           { m.AddUint64(k, uint64(v)) }
+func (m *ObjectEncoder) AddUint32(k string, v uint32)       { m.AddInt64(k, int64(v)) }
+func (m *ObjectEncoder) AddUint16(k string, v uint16)       { m.AddInt64(k, int64(v)) }
+func (m *ObjectEncoder) AddUint8(k string, v uint8)         { m.AddInt64(k, int64(v)) }
+func (m *ObjectEncoder) AddUintptr(k string, v uintptr)     { m.AddUint64(k, uint64(v)) }
+
+// ArrayEncoder implements zapcore.ArrayEncoder.
+type ArrayEncoder struct {
```

Comment:
Does it need to be exported?

```suggestion
type arrayEncoder struct {
```

Reply:
ObjectEncoder is an interface by zap which expects this to be exported. One example where a user may use this is https://pkg.go.dev/go.uber.org/zap#example-Object